src/components/CouponDetails/index.jsx

```diff
@@ -483,11 +490,15 @@ class CouponDetails extends React.Component {
 
     this.resetCodeActionStatus();
 
-    if (stateKey) {
+    if (stateKey || doesCodeActionHaveErrors) {
       this.setState({
-        [stateKey]: true,
-        refreshIndex: this.state.refreshIndex + 1, // force new table instance
-        selectedCodes: [],
+        ...(stateKey ? {
```

Comment:
Can we not just add `doesCodeActionHaveErrors` in the state? It would be simpler?